### ISSUE TITLE
arithmetic error COMMON-LISP:FLOATING-POINT-INVALID-OPERATION

### ISSUE THREAD

**Issue by Ambrevar** [coalton-lang/coalton#658]
On 0a95e326e567c09c06be2dcc2073f1aa09736a6c I get this when I run the tests:

```log
[...]
  TEST-SLICE-CHUNKED-ITERATION............................................[ OK ]
  TEST-FLOAT
Unhandled COMMON-LISP:FLOATING-POINT-INVALID-OPERATION in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                                    {1003378073}>:
  arithmetic error COMMON-LISP:FLOATING-POINT-INVALID-OPERATION signalled

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1003378073}>
0: ("bogus stack frame")
1: (BIG-FLOAT::|INSTANCE/INTO SINGLE-FLOAT BIG-FLOAT-COALTON-LIBRARY/CLASSES:INTO| #<SINGLE-FLOAT quiet NaN>)
2: (COALTON-LIBRARY/LIST::|INSTANCE/FUNCTOR LIST-COALTON-LIBRARY/CLASSES:MAP| #<COALTON-IMPL/CODEGEN/FUNCTION-ENTRY::FUNCTION-ENTRY :ARITY 1> (#<SINGLE-FLOAT quiet NaN> 0.0 1.0 3.1415927 2.7182817 -3.1415927 1.4142135 1.8137995 0.5 10.0 100.0))
3: (COALTON-LIBRARY/LIST::|INSTANCE/FUNCTOR LIST-COALTON-LIBRARY/CLASSES:MAP| #<COALTON-IMPL/CODEGEN/FUNCTION-ENTRY::FUNCTION-ENTRY :ARITY 1> (#.SB-EXT:SINGLE-FLOAT-POSITIVE-INFINITY #<SINGLE-FLOAT quiet NaN> 0.0 1.0 3.1415927 2.7182817 -3.1415927 1.4142135 1.8137995 0.5 10.0 100.0))
4: ((COMMON-LISP:LABELS TEST-FLOAT :IN TEST-FLOAT))
5: ((COMMON-LISP:LABELS FIASCO::RUN-TEST-BODY :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS))
6: (FIASCO::CALL-WITH-TEST-HANDLERS #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS) {100F0586DB}>)
7: (FIASCO::PRETTY-RUN-TEST #<test TEST-FLOAT> #<FUNCTION (COMMON-LISP:LABELS TEST-FLOAT :IN TEST-FLOAT) {53CCD61B}>)
8: ((COMMON-LISP:LABELS #:BODY-SYM0 :IN TEST-FLOAT))
9: (TEST-FLOAT)
10: ((COMMON-LISP:LABELS FIASCO-SUITES::COALTON-TESTS :IN FIASCO-SUITES::COALTON-TESTS))
11: ((COMMON-LISP:LABELS FIASCO::RUN-TEST-BODY :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS))
12: (FIASCO::CALL-WITH-TEST-HANDLERS #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS) {100F2AF36B}>)
13: (FIASCO::PRETTY-RUN-TEST #<test FIASCO-SUITES::COALTON-TESTS :tests 64> #<FUNCTION (COMMON-LISP:LABELS FIASCO-SUITES::COALTON-TESTS :IN FIASCO-SUITES::COALTON-TESTS) {53C4101B}>)
14: ((COMMON-LISP:LABELS #:BODY-SYM1 :IN FIASCO-SUITES::COALTON-TESTS))
15: (FIASCO-SUITES::COALTON-TESTS)
16: (FIASCO:RUN-SUITE-TESTS #<test FIASCO-SUITES::COALTON-TESTS :tests 64> :VERBOSE COMMON-LISP:NIL :STREAM #<COMMON-LISP:SYNONYM-STREAM :SYMBOL SB-SYS:*STDOUT* {1000013513}> :INTERACTIVE COMMON-LISP:T)
17: (FIASCO:RUN-TESTS (:COALTON-TESTS :QUIL-COALTON-TESTS :THIH-COALTON-TESTS :COALTON-JSON-TESTS) :DESCRIBE-FAILURES COMMON-LISP:T :VERBOSE COMMON-LISP:NIL :STREAM #<COMMON-LISP:SYNONYM-STREAM :SYMBOL SB-SYS:*STDOUT* {1000013513}> :INTERACTIVE COMMON-LISP:T)
18: ((:METHOD ASDF/ACTION:PERFORM (ASDF/LISP-ACTION:TEST-OP (COMMON-LISP:EQL #<ASDF/SYSTEM:SYSTEM "coalton/tests">))) #<unused argument> #<unused argument>) [fast-method]
19: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton/tests">)
20: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
21: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (COMMON-LISP:T COMMON-LISP:T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton/tests">) [fast-method]
22: ((:METHOD ASDF/PLAN:PERFORM-PLAN (COMMON-LISP:T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {100776A0C3}>) [fast-method]
23: ((COMMON-LISP:FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
24: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (COMMON-LISP:T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {100776A0C3}>) [fast-method]
25: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton"> :PLAN-CLASS COMMON-LISP:NIL :PLAN-OPTIONS COMMON-LISP:NIL) [fast-method]
26: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton">)
27: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
28: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton">) [fast-method]
29: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:TEST-OP "coalton")
30: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
31: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) ASDF/LISP-ACTION:TEST-OP "coalton") [fast-method]
32: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE) {100775AD7B}> :OVERRIDE COMMON-LISP:T :KEY COMMON-LISP:NIL :OVERRIDE-CACHE COMMON-LISP:T :OVERRIDE-FORCING COMMON-LISP:NIL)
33: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
34: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE) {10077590AB}> :OVERRIDE COMMON-LISP:NIL :KEY COMMON-LISP:NIL :OVERRIDE-CACHE COMMON-LISP:NIL :OVERRIDE-FORCING COMMON-LISP:NIL)
35: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) ASDF/LISP-ACTION:TEST-OP "coalton") [fast-method]
36: (ASDF/OPERATE:TEST-SYSTEM "coalton")
37: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ASDF/OPERATE:TEST-SYSTEM "coalton") #<NULL-LEXENV>)
38: (COMMON-LISP:EVAL (ASDF/OPERATE:TEST-SYSTEM "coalton"))
39: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(require :asdf)") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/4s3d7axl86d7llr3fgz8m06sd6324346-sbcl-coalton-0.0.1-3.0a95e32/share/common-lisp/sbcl/coalton/coalton.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/4s3d7axl86d7llr3fgz8m06sd6324346-sbcl-coalton-0.0.1-3.0a95e32/share/common-lisp/sbcl/coalton/examples/coalton-json/coalton-json.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/4s3d7axl86d7llr3fgz8m06sd6324346-sbcl-coalton-0.0.1-3.0a95e32/share/common-lisp/sbcl/coalton/examples/quil-coalton/quil-coalton.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/4s3d7axl86d7llr3fgz8m06sd6324346-sbcl-coalton-0.0.1-3.0a95e32/share/common-lisp/sbcl/coalton/examples/small-coalton-programs/small-coalton-programs.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/4s3d7axl86d7llr3fgz8m06sd6324346-sbcl-coalton-0.0.1-3.0a95e32/share/common-lisp/sbcl/coalton/examples/thih/thih-coalton.asd\"))") (:EVAL . "(when (uiop:file-exists-p \"coalton-tests.asd\") (asdf:load-asd (truename \"coalton-tests.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"coalton-test.asd\") (asdf:load-asd (truename \"coalton-test.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"tests.asd\") (asdf:load-asd (truename \"tests.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"test.asd\") (asdf:load-asd (truename \"test.asd\")))") (:EVAL . "(asdf:test-system \"coalton\")") (:QUIT)))
40: (SB-IMPL::TOPLEVEL-INIT)
41: ((COMMON-LISP:FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
42: ((COMMON-LISP:FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
43: (SB-IMPL::%START-LISP)
```

It used to work on 012f6c8db6d73df16f7729090a12a929fb82db17.

**Comment by eliaslfox** [comment]
What is your sbcl version, os, and cpu architecture?

**Comment by Ambrevar** [comment]
SBCL 2.2.6
OS: Guix
CPU: x86_64.

**Comment by eliaslfox** [comment]
This looks like a failure converting a single float nan to a big float. Could you try running the following:

```
(ql:quickload :coalton/library/big-float)
(in-package :coalton-user)
(coalton (the coalton-library/big-float:Big-Float (the Single-Float math:nan)))
```

On sbcl 2.2.4 I get: `0e-1`

**Comment by Ambrevar** [comment]
```lisp
(asdf:load-system :coalton/library/big-float)
WARNING: System definition file #P"/gnu/store/3clyw47fqk6d8ngrzv9c8bl78v655gp9-sbcl-trivial-garbage-0.21/share/common-lisp/sbcl/trivial-garbage/trivial-garbage.asd" contains definition for system "trivial-garbage-tests". Please only define "trivial-garbage" and secondary systems with a name starting with "trivial-garbage/" (e.g. "trivial-garbage/test") in that file.
STYLE-WARNING:
   Generic function FSET:ITERATOR clobbers an earlier FTYPE proclamation
   (FUNCTION (T &KEY &ALLOW-OTHER-KEYS) (VALUES FUNCTION &REST T)) for the same
   name with (FUNCTION (T &KEY &ALLOW-OTHER-KEYS) *).
COMMON-LISP:WARNING: redefining SB-MPFR::MPFR_GET_Q in DEFUN
T
* (in-package :coalton-user)
#<COMMON-LISP:PACKAGE "COALTON-USER">
* (coalton (the coalton-library/big-float:Big-Float (into (the Single-Float math:nan))))

debugger invoked on a COMMON-LISP:FLOATING-POINT-INVALID-OPERATION in thread
#<THREAD "main thread" RUNNING {100B6E8073}>:
  arithmetic error COMMON-LISP:FLOATING-POINT-INVALID-OPERATION signalled

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [ABORT] Exit debugger, returning to top level.

("bogus stack frame")
0] 
```

**Comment by eliaslfox** [comment]
I pushed what I think is a fix in https://github.com/coalton-lang/coalton/commit/0ed40253c6948fb1a1e942aa9e6c4aba08ad66f7. I tried on sbcl 2.2.6, but I wasn't able to reproduce the issue.

**Comment by Ambrevar** [comment]
Sadly I still get the same issue.

**Comment by stylewarning** [comment]
> Sadly I still get the same issue.

Same stack frames too?

**Comment by Ambrevar** [comment]
I think so:

```log
  TEST-FLOAT
Unhandled COMMON-LISP:FLOATING-POINT-INVALID-OPERATION in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                                    {1002A58073}>:
  arithmetic error COMMON-LISP:FLOATING-POINT-INVALID-OPERATION signalled

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1002A58073}>
0: ("bogus stack frame")
1: (BIG-FLOAT::|INSTANCE/INTO SINGLE-FLOAT BIG-FLOAT-COALTON-LIBRARY/CLASSES:INTO| #<SINGLE-FLOAT quiet NaN>)
2: (COALTON-LIBRARY/LIST::|INSTANCE/FUNCTOR LIST-COALTON-LIBRARY/CLASSES:MAP| #<COALTON-IMPL/CODEGEN/FUNCTION-ENTRY::FUNCTION-ENTRY :ARITY 1> (#<SINGLE-FLOAT quiet NaN> 0.0 1.0 3.1415927 2.7182817 -3.1415927 1.4142135 1.8137995 0.5 10.0 100.0))
3: (COALTON-LIBRARY/LIST::|INSTANCE/FUNCTOR LIST-COALTON-LIBRARY/CLASSES:MAP| #<COALTON-IMPL/CODEGEN/FUNCTION-ENTRY::FUNCTION-ENTRY :ARITY 1> (#.SB-EXT:SINGLE-FLOAT-POSITIVE-INFINITY #<SINGLE-FLOAT quiet NaN> 0.0 1.0 3.1415927 2.7182817 -3.1415927 1.4142135 1.8137995 0.5 10.0 100.0))
4: ((COMMON-LISP:LABELS TEST-FLOAT :IN TEST-FLOAT))
5: ((COMMON-LISP:LABELS FIASCO::RUN-TEST-BODY :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS))
6: (FIASCO::CALL-WITH-TEST-HANDLERS #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS) {100F2286DB}>)
7: (FIASCO::PRETTY-RUN-TEST #<test TEST-FLOAT> #<FUNCTION (COMMON-LISP:LABELS TEST-FLOAT :IN TEST-FLOAT) {53CD1CFB}>)
8: ((COMMON-LISP:LABELS #:BODY-SYM0 :IN TEST-FLOAT))
9: (TEST-FLOAT)
10: ((COMMON-LISP:LABELS FIASCO-SUITES::COALTON-TESTS :IN FIASCO-SUITES::COALTON-TESTS))
11: ((COMMON-LISP:LABELS FIASCO::RUN-TEST-BODY :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS))
12: (FIASCO::CALL-WITH-TEST-HANDLERS #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN FIASCO::RUN-TEST-BODY-IN-HANDLERS) {100F4876FB}>)
13: (FIASCO::PRETTY-RUN-TEST #<test FIASCO-SUITES::COALTON-TESTS :tests 65> #<FUNCTION (COMMON-LISP:LABELS FIASCO-SUITES::COALTON-TESTS :IN FIASCO-SUITES::COALTON-TESTS) {53C4572B}>)
14: ((COMMON-LISP:LABELS #:BODY-SYM1 :IN FIASCO-SUITES::COALTON-TESTS))
15: (FIASCO-SUITES::COALTON-TESTS)
16: (FIASCO:RUN-SUITE-TESTS #<test FIASCO-SUITES::COALTON-TESTS :tests 65> :VERBOSE COMMON-LISP:NIL :STREAM #<COMMON-LISP:SYNONYM-STREAM :SYMBOL SB-SYS:*STDOUT* {1000013513}> :INTERACTIVE COMMON-LISP:T)
17: (FIASCO:RUN-TESTS (:COALTON-TESTS :QUIL-COALTON-TESTS :THIH-COALTON-TESTS :COALTON-JSON-TESTS) :DESCRIBE-FAILURES COMMON-LISP:T :VERBOSE COMMON-LISP:NIL :STREAM #<COMMON-LISP:SYNONYM-STREAM :SYMBOL SB-SYS:*STDOUT* {1000013513}> :INTERACTIVE COMMON-LISP:T)
18: ((:METHOD ASDF/ACTION:PERFORM (ASDF/LISP-ACTION:TEST-OP (COMMON-LISP:EQL #<ASDF/SYSTEM:SYSTEM "coalton/tests">))) #<unused argument> #<unused argument>) [fast-method]
19: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton/tests">)
20: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
21: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (COMMON-LISP:T COMMON-LISP:T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton/tests">) [fast-method]
22: ((:METHOD ASDF/PLAN:PERFORM-PLAN (COMMON-LISP:T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {10069A3BB3}>) [fast-method]
23: ((COMMON-LISP:FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
24: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (COMMON-LISP:T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {10069A3BB3}>) [fast-method]
25: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton"> :PLAN-CLASS COMMON-LISP:NIL :PLAN-OPTIONS COMMON-LISP:NIL) [fast-method]
26: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton">)
27: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
28: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) #<ASDF/LISP-ACTION:TEST-OP > #<ASDF/SYSTEM:SYSTEM "coalton">) [fast-method]
29: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:TEST-OP "coalton")
30: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
31: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) ASDF/LISP-ACTION:TEST-OP "coalton") [fast-method]
32: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE) {10069947DB}> :OVERRIDE COMMON-LISP:T :KEY COMMON-LISP:NIL :OVERRIDE-CACHE COMMON-LISP:T :OVERRIDE-FORCING COMMON-LISP:NIL)
33: ((COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE))
34: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (COMMON-LISP:LAMBDA COMMON-LISP:NIL :IN ASDF/OPERATE:OPERATE) {1006992B0B}> :OVERRIDE COMMON-LISP:NIL :KEY COMMON-LISP:NIL :OVERRIDE-CACHE COMMON-LISP:NIL :OVERRIDE-FORCING COMMON-LISP:NIL)
35: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (COMMON-LISP:T COMMON-LISP:T)) ASDF/LISP-ACTION:TEST-OP "coalton") [fast-method]
36: (ASDF/OPERATE:TEST-SYSTEM "coalton")
37: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ASDF/OPERATE:TEST-SYSTEM "coalton") #<NULL-LEXENV>)
38: (COMMON-LISP:EVAL (ASDF/OPERATE:TEST-SYSTEM "coalton"))
39: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(require :asdf)") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/ai1gi8h0xsp2f84y9qiw38ycbwk7vdvl-sbcl-coalton-0.0.1-3.0ed4025/share/common-lisp/sbcl/coalton/coalton.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/ai1gi8h0xsp2f84y9qiw38ycbwk7vdvl-sbcl-coalton-0.0.1-3.0ed4025/share/common-lisp/sbcl/coalton/examples/coalton-json/coalton-json.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/ai1gi8h0xsp2f84y9qiw38ycbwk7vdvl-sbcl-coalton-0.0.1-3.0ed4025/share/common-lisp/sbcl/coalton/examples/quil-coalton/quil-coalton.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/ai1gi8h0xsp2f84y9qiw38ycbwk7vdvl-sbcl-coalton-0.0.1-3.0ed4025/share/common-lisp/sbcl/coalton/examples/small-coalton-programs/small-coalton-programs.asd\"))") (:EVAL . "(asdf:load-asd (truename \"/gnu/store/ai1gi8h0xsp2f84y9qiw38ycbwk7vdvl-sbcl-coalton-0.0.1-3.0ed4025/share/common-lisp/sbcl/coalton/examples/thih/thih-coalton.asd\"))") (:EVAL . "(when (uiop:file-exists-p \"coalton-tests.asd\") (asdf:load-asd (truename \"coalton-tests.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"coalton-test.asd\") (asdf:load-asd (truename \"coalton-test.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"tests.asd\") (asdf:load-asd (truename \"tests.asd\")))") (:EVAL . "(when (uiop:file-exists-p \"test.asd\") (asdf:load-asd (truename \"test.asd\")))") (:EVAL . "(asdf:test-system \"coalton\")") (:QUIT)))
40: (SB-IMPL::TOPLEVEL-INIT)
41: ((COMMON-LISP:FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
42: ((COMMON-LISP:FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
43: (SB-IMPL::%START-LISP)
```

**Comment by eliaslfox** [comment]
This should be fixed by #839 which disables all float traps.
